### PR TITLE
Added config documentation

### PR DESCRIPTION
**Pull request by Pancia**

### CODE REVIEW

File: reference/reference.adoc [hunk 1]
```diff
@@ -646,6 +646,71 @@ exceptions, since the default server implementation sends back a hard error.
 the app state in any way it sees fit to represent the handling of the error (e.g. change UI state to show an error
 dialog, reload the page, etc.).
 
+=== Configuration
+
```
Comment (awkay) [inline reference/reference.adoc:694]: Let's add a section on the fact that config is done through a SS component, and that you can inject it into the parser

Comment (awkay) [inline reference/reference.adoc:694]: There should be some discussion about how the overall config is structured before talking about specific files:
- Motivation: using EDN for making easy to user
- Choosing config for development vs staging vs CI vs production
- Using config from an injected component

The fact that the same filename can be used on different machines makes it possible to support all of these concerns without doing ANY code or CLI config; however, the -Dconfig option allows you to easily use the source-based configs as well (e.g. include configs in source code and select from CLI).

The disk path configured in CODE is meant to be the DEFAULT SYSTEM config location...it is never edited to just change environment. instead as you run the thing on different machines, it picks up that machine's version of that file. E.g. /usr/local/etc/app.edn is diff on each machine.

The CLI override is meant to handle the case where you want to temporarily (for whatever reason) use an alternate.

Also: the defaults.edn should be explained. When is it merged? How is it used? You say nothing about how it is providing a baseline for all configs. I'd specifically mention the merge order/behavior.

**Review by egracer — CHANGES_REQUESTED** [review]
Can you add something mentioning how config can contain arbitrary keywords and values that can be accessed in the environment of server-side mutations?

**Comment by Pancia** [comment]
[Ref Guide - Config](https://trello.com/c/moIlszj2/3-ref-guide-config)

File: reference/reference.adoc (hunk 1 above)
Comment (Pancia) [reply reference/reference.adoc:694]: https://github.com/untangled-web/untangled/pull/5/files#diff-17de57c76e8350ed0f0c2c9339d32571R654

**Pull request merged**